File: typescript/src/routes/logs.ts
import express, { Request, Response, Router } from "express";
import fs from "fs";
import path from "path";
import config from "../config";
import datastore from "../datastore";

const router: Router = express.Router();

router.get("/logs", (req: Request, res: Response) => {
  res.json(datastore.tail());
});

export default router;

File: python/src/nucleoidai/routes/logs.py
"""Logs API routes for Nucleoid AI."""

from typing import List, Dict, Any
from fastapi import APIRouter, Request

from ..datastore import DataStore

router = APIRouter()
datastore = DataStore()


@router.get("/logs")
async def get_logs(request: Request) -> List[Dict[str, Any]]:
    """Get recent execution logs."""
    logs = datastore.tail()
    
    # Convert Data objects to dictionaries for JSON serialization
    result = []
    for log in logs:
        if hasattr(log, '__dict__'):
            log_dict = log.__dict__.copy()
            # Convert datetime to string if present
            if 'date' in log_dict and log_dict['date']:
                log_dict['date'] = log_dict['date'].isoformat()
            result.append(log_dict)
        else:
            result.append(log)
    
    return result